### PR TITLE
[5145] - Make raw HESA collection data accessible to System Admins

### PR DESCRIPTION
**Pull request by d-a-v-e**
### Context

We want to make raw HESA collection data accessible from trainee records for admin users.

Admins often need to look at a trainee record, and compare that with what was submitted to HESA. Currently that can be done by looking up the slug in Blazer - but this is slow. It would be easier if the raw collection data was accessible from the record. 

[Link to page in prototype](https://register-prototype.herokuapp.com/record/08283e35-45f4-47dc-b8a7-0ddd305e31c1/admin)

This was [added to the prototype in this pr](https://github.com/DFE-Digital/register-trainee-teachers-prototype/pull/611). You can try in the prototype by setting the user type to admin and viewing a record.

It's expected we may add more features / rows / metadata to this admin tab in the future.

### Changes proposed in this pull request

Adds HesaStudentSummary view component and corresponding page visible only to admins

### Guidance to review

* real data is needed to test retrieval of HESA data so please make use of the productiondata deploy (or use a local copy)

### Important business

- [ ] ~Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?~
- [ ] ~Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?~
- [ ] ~Do we need to send any updates to DQT as part of the work in this PR?~
- [ ] ~Does this PR need an ADR?~

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Review by edwardhorsford — CHANGES_REQUESTED** [review]
Looking good @d-a-v-e !

Some comments / suggestions:

* ~Reckon the order of rows should be chronological - so c21 then c22~
* Only show a collection row if we have data for it. So if we only have C22, don't show a row for C21 (I think register currently only ever has data for a single collection for a trainee - but hopefully that will change soon)
* I think the headings should likely be h3s? - the summary card itself has an h2, so this should be the next level down
* Ideally the headings would be correctly pluralised based on the contents. But if you don’t fancy bothering, just use the plural form, so Degrees , not Degree(s)
* For non hesa trainees, don’t render the hesa summary card at all

What date is shown in the heading of the details? I’d expect it to be the last modified date or last time we received new data from HESA. But 15 November 2022 is well before the hesa updated at date.

File: app/views/trainees/admins/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<section class="app-summary-card govuk-!-margin-bottom-6">
+   <header class="app-summary-card__header">
+      <h2 class="app-summary-card__title">
+         HESA
+      </h2>
+   </header>
+   <div class="app-summary-card__body">
+      <dl class="govuk-summary-list" style="overflow:hidden;">
+         <%= render HesaStudentSummary::View.new(trainee:, collection: "C22053") %>
+         <%= render HesaStudentSummary::View.new(trainee:, collection: "C21053") %>
```
Comment (edwardhorsford) [inline app/views/trainees/admins/show.html.erb:10]: Feels like these should be dynamic based on what data we have?

**Comment by edwardhorsford** [comment]
Disregard the chronological request - @claire-hughez thinks the newest should be first.

**Comment by claire-hughez** [comment]
> Disregard the chronological request - @claire-hughez thinks the newest should be first.

Because I'm a tyrant....

Nah, because I think for Support (and for us potentially?), it will make more sense for the most recent HESA collection to be first, as that's what will be most relevant for the actual trainee record they see.

**Comment by d-a-v-e** [comment]
> * Only show a collection row if we have data for it. So if we only have C22, don't show a row for C21 (I think register currently only ever has data for a single collection for a trainee - but hopefully that will change soon)

done

> * I think the headings should likely be h3s? - the summary card itself has an h2, so this should be the next level down

done

> * Ideally the headings would be correctly pluralised based on the contents. But if you don’t fancy bothering, just use the plural form, so Degrees , not Degree(s)

done

> * For non hesa trainees, don’t render the hesa summary card at all

and done!

**Comment by HettieS** [comment]
[[S] Admins: Make raw HESA collection data accessible from trainee records](https://trello.com/c/IaPTEiiq/5145-s-admins-make-raw-hesa-collection-data-accessible-from-trainee-records)

**Review by HettieS — COMMENTED** [review]
Looks good, I've left a few questions. I'll give it a review in prod data env and will approve if all looks good :)

File: app/controllers/application_controller_dev.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@ module ApplicationControllerDev
   # Current User set to first admin profile
   def current_user
     @current_user ||= begin
-      user = User.where(system_admin: true).first
+      user = User.where(system_admin: false).first
```
Comment (HettieS) [inline app/controllers/application_controller_dev.rb:11]: I'm not sure what this file is doing?

File: app/components/hesa_student_summary/view.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    end
+
+    def degrees
+      return [] if hesa_trainee&.degrees.blank?
+
+      @degrees ||= hesa_trainee.degrees.map { |degree| summarise(degree) }.flatten
+    end
+
+    def degree_count
+      return if degrees.empty?
```
[the remark this answers is not in the document]
Comment (d-a-v-e) [reply app/components/hesa_student_summary/view.rb:31]: only as I'm expecting an array either way

File: app/components/tab_navigation/view.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@ class View < GovukComponent::Base
     attr_reader :items
 
     def initialize(items:)
-      @items = items
+      @items = items&.compact
```
[the remark this answers is not in the document]
Comment (d-a-v-e) [reply app/components/tab_navigation/view.rb:8]: probably not, but I wasn't sure if this ever gets instantiated without any items, so just a precaution.

File: app/components/hesa_student_summary/view.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      @placements ||= hesa_trainee.placements.map { |placement| summarise(placement) }.flatten
+    end
+
+    def placement_count
+      return if placements.empty?
+
+      hesa_trainee.placements.count
+    end
+
+    # 4 December 2022 at 1:07pm
```
[the remark this answers is not in the document]
Comment (d-a-v-e) [reply app/components/hesa_student_summary/view.rb:48]: I've left this in as the `strftime` is a bit cryptic

File: app/views/trainees/admins/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<% if trainee.hesa_id %>
```
[the remark this answers is not in the document]
Comment (d-a-v-e) [reply app/views/trainees/admins/show.html.erb:1]: good shout, I'll check for a record instead

**Review by HettieS — APPROVED** [review]

**Pull request merged**